legend-engine-application-query/src/test/java/org/finos/legend/engine/application/query/api/TestQueryStoreManager.java

```diff
@@ -734,4 +734,28 @@ public void testGetLightQueriesSortWithLastUpdatedAt() throws Exception
         Query lightQuery = queries.get(0);
         Assert.assertEquals("2", lightQuery.id);
     }
+
+    @Test
+    public void testGetQueriesWithSearchByQueryId() throws Exception
+    {
+        String currentUser = "testUser";
+        queryStoreManager.createQuery(TestQueryBuilder.create("11", "query_a", currentUser).build(), currentUser);
+        queryStoreManager.createQuery(TestQueryBuilder.create("12", "query_b", currentUser).build(), currentUser);
+        queryStoreManager.createQuery(TestQueryBuilder.create("13", "query_c", currentUser).build(), currentUser);
+        Assert.assertEquals(3, queryStoreManager.getQueries(new TestQuerySearchSpecificationBuilder().build(), currentUser).size());
```

Comment:
I would like us to be a little more refined here, allowing search id like this would harbor mumbo jumbo in our search results, I think the best way is to check if the search term is of a particular format (i.e. (uuid)[https://en.wikipedia.org/wiki/Universally_unique_identifier] or something slightly looser), then we would proceed to do the ID matching